presto-main/src/main/java/com/facebook/presto/sql/planner/DeterminismEvaluator.java

```diff
@@ -47,7 +48,8 @@ protected Void visitFunctionCall(FunctionCall node, AtomicBoolean deterministic)
             // TODO: total hack to figure out if a function is deterministic. martint should fix this when he refactors the planning code
             if (node.getName().equals(QualifiedName.of("rand")) ||
                     node.getName().equals(QualifiedName.of("random")) ||
-                    node.getName().equals(QualifiedName.of("shuffle"))) {
+                    node.getName().equals(QualifiedName.of("shuffle")) ||
+                    node.getName().equals(QualifiedName.of(GROUPING))) {
```

Comment:
There's already a hack alert at the top of the if statement so I don't know if adding another one will help. Or am I misunderstanding and you want me to explain something else?

The reason why I added this is because otherwise the `PredicatePushDown` optimizer tries to push the `grouping()` function past the `GroupId` node, which is not allowed. Maybe there's a better way to stop this from happening but I really don't know `PredicatePushDown` code and I'm not sure.

The query I'm debugging is this one
```
SELECT a, b, sum(c), grouping(a, b)
FROM (VALUES ('h', 'j', 11), ('k', 'l', 7)) AS t (a, b, c)
GROUP BY GROUPING SETS ( (a), (b))
HAVING grouping(a, b) > 1
```
and the plan is
```
ProjectNode <- FilterNode (grouping) <- AggregationNode <- GroupIdNode
```
I'm trying to come up with other queries where a push down of `grouping()` will be beneficial but I'm failing to come up with something.